src/UseCases/ReadAddressChange/ReadAddressChangeUseCase.php

```diff
@@ -18,7 +18,7 @@ public function __construct( AddressChangeRepository $addressChangeRepository )
 	public function getAddressChangeByUuids( string $currentIdentifier, string $previousIdentifier ): ?AddressChangeData {
 		$addressChange = $this->addressChangeRepository->getAddressChangeByUuids( $currentIdentifier, $previousIdentifier );
 
-		if ( !$addressChange ) {
+		if ( !$addressChange || !$addressChange->getPreviousIdentifier() ) {
```

Comment:
This changes the meaning and logic - If there is an address change, but with empty previous identifier, this will return null. Every donation and membership starts with an Address change with an empty previous identifier. Only filling out an address change will turn the current into the previous. This change would completely break the address changes, because no address changes would be returned.

Lucky for us, the nullable `getPreviousIdentifier` getter is is a faulty leftover from a refactoring when we made the property inside the class non-nullable. So please remove this change and make `AddressChange::getPreviousIdentifier` non-nullable.

Reply:
Thank you.